internal/cmd/db_show.go

```diff
@@ -85,55 +83,31 @@ var showCmd = &cobra.Command{
 		copy(regions, db.Regions)
 		sort.Strings(regions)
 
-		versions := [](chan string){}
-		urls := []string{}
-		inspectRet := InspectInfo{}
-		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
-		defer cancel()
-		g, ctx := errgroup.WithContext(ctx)
-		results := make(chan *InspectInfo, len(instances))
-		for idx, instance := range instances {
-			urls = append(urls, getInstanceUrl(config, &db, &instance))
-			versions = append(versions, make(chan string, 1))
-			go func(idx int, client *turso.Client, config *settings.Settings, db *turso.Database, instance *turso.Instance) {
-				versions[idx] <- fetchInstanceVersion(client, config, db, instance)
-			}(idx, client, config, &db, &instance)
-			loopInstance := instance
-			g.Go(func() error {
-				url := getInstanceHttpUrl(config, &db, &loopInstance)
-				ret, err := inspect(ctx, url, token, loopInstance.Region, verboseFlag)
-				if err != nil {
-					return err
-				}
-				results <- ret
-				return nil
-			})
-		}
-		if err := g.Wait(); err != nil {
-			if errors.Is(err, context.DeadlineExceeded) {
-				return fmt.Errorf("timeout while inspecting database. It's possible that this database is too old and does not support inspecting or one of the instances is not reachable")
-			}
+		instancesInfo := getInstancesInfo(client, instances, config, db, token)
+		if err != nil {
 			return err
 		}
-		for range instances {
-			ret := <-results
-			inspectRet.Accumulate(ret)
-		}
 		data := [][]string{}
 		for idx, instance := range instances {
-			version := <-versions[idx]
+			version := <-instancesInfo.versions[idx]
 			if showInstanceUrlsFlag {
-				data = append(data, []string{instance.Name, instance.Type, instance.Region, version, urls[idx]})
+				data = append(data, []string{instance.Name, instance.Type, instance.Region, version, instancesInfo.urls[idx]})
 			} else {
 				data = append(data, []string{instance.Name, instance.Type, instance.Region, version})
 			}
 		}
+		var size string
+		if instancesInfo.inspectInfo == nil {
+			size = fmt.Sprintf("fetching size failed: %s", err)
```

Comment:
The value of err here seems to be something random. Probably nil if check in line 87 didn't return.